### ISSUE TITLE
Conflicts on processed data and load_dataset when train_controlnet_sd

### ISSUE THREAD

**Issue by fangchuan** [OrangeSodahub/SceneCraft#14]
Hi, I processed the Hypersim dataset following the readme. The gennerated json file of a single scene contains data in the below:

`{"scene_id": "ai_001_001", "id": "frame.0000", "target": "./datasets/hypersim/scenes/ai_001_001/images/scene_cam_00_final_preview/frame.0000.gamma.jpg", "source": "./datasets/hypersim/semantic_images/ai_001_001/frame.0000.npz", "caption": "a bathroom with a tub, sink, and two windows", "item_id": [0, 1, 2, 3, 9, 13, 15, 19, 27, 34, 35, 36, 38, 40], "max_depth": 4.26171875}
{"scene_id": "ai_001_001", "id": "frame.0001", "target": "./datasets/hypersim/scenes/ai_001_001/images/scene_cam_00_final_preview/frame.0001.gamma.jpg", "source": "./datasets/hypersim/semantic_images/ai_001_001/frame.0001.npz", "caption": "a bath room with a bath tub a sink and a window", "item_id": [1, 2, 3, 9, 13, 15, 19, 27, 34, 35, 36, 38, 40], "max_depth": 3.904296875}`

However, when I want to validate if current preparation is enough to run `scenecraft/finetune/train_controlnet_sd.py`, it comes to several problems:
1. load the processed data through load_dataset() func needs tobe revised: From https://github.com/OrangeSodahub/SceneCraft/blob/a4daebfc56cf663c463729944960ef5f5a31eab3/scenecraft/finetune/train_controlnet_sd.py#L685.   to 
`raw_dataset = load_dataset('json', data_files='/mnt/nas_3dv/hdd1/fangchuan/SceneCraft/datasets/hypersim/semantic_images/train.jsonl')`
2.  The expected data columns are conflicted:  https://github.com/OrangeSodahub/SceneCraft/blob/a4daebfc56cf663c463729944960ef5f5a31eab3/scenecraft/finetune/train_controlnet_sd.py#L699 ,  while `generate_json()` only gives these data fileds: https://github.com/OrangeSodahub/SceneCraft/blob/a4daebfc56cf663c463729944960ef5f5a31eab3/scenecraft/renderer.py#L892-L899

I guess the code become conflicted because serval updated commits, if so, could you help me clean the code or give me more clues to setup the correct training setting? @OrangeSodahub Many thanks in advance!

**Comment by fangchuan** [comment]
Great! Thank you very much. I fix the previous problem to run `train_controlnet_sd`, but further run into another problem when I want evaluate the given ckpt using `inference_controlnet_sd`, it seems that there is no `embeddings` filed in dataloader:

<img width="1112" alt="Image" src="https://github.com/user-attachments/assets/110607a2-6c27-457d-a54b-82fed9f06bb7" />

As the FinetuneDataset donot give the embeddings
https://github.com/OrangeSodahub/SceneCraft/blob/a4daebfc56cf663c463729944960ef5f5a31eab3/scenecraft/data/dataset.py#L433-L441

**Comment by fangchuan** [comment]
Hi, thank your suggestion. I finally resolved all problems and run the nerf sds fitting successfully. But the result seems wired, same as the issue in #12 . 

My trainer config:

`scenecraft_method_nerfacto_huge2 = MethodSpecification(
    config=SceneCraftTrainerConfig(
        method_name="nerfacto-huge2",
        experiment_name="nerf",
        steps_per_eval_batch=50,
        steps_per_eval_image=50,
        steps_per_save=200,
        max_num_iterations=15000,
        save_only_latest_checkpoint=False,
        mixed_precision=True,
        gradient_accumulation_steps=1,
        pipeline=SceneCraftPipelineConfig(
            enable_modelAB=False,
            async_mode=True,
            full_image_every=1,
            diffusion_steps=20,
            scheduler_type="unipc",
            guidance_depth=False,
            fix_init_noise=False,
            controlnet_conditioning_scale=[3.5, 1.5],
            datamanager=SceneCraftDataManagerConfig(
                dataparser=DataParserConfig(
                    # dataset_type="scannetpp",
                    # data=Path("outputs/scannetpp"),
                    # scene_id="0a7cc12c0e",
                    dataset_type="hypersim",
                    data=Path("outputs/hypersim"),
                    scene_id="ai_001_005",
                    # dataset_type="custom",
                    # data=Path("outputs/custom"),
                    # scene_id="exp5",
                ),
                condition_type="one_hot",
                train_num_images_to_sample_from=2,
                train_num_times_to_repeat_images=0,
                nerf_batch_size=1,
                # camera_res_scale_factor=0.5754, # [584, 876] -> [336, 504]
                # guide_camera_res_scale_factor=0.8768,
                camera_res_scale_factor=0.4375,
                guide_camera_res_scale_factor=0.6875,
                guide_buffer_size=250,
            ),
            downscale_factor=0.5,
            prompt="This is one view of a bedroom painted by VanGogh.",
            time_schedule=[
                [600, -1, -1], [1000, 0.8, 0.98], [1300, 0.7, 0.8], [1600, 0.6, 0.7], [1900, 0.5, 0.6], [2200, 0.2, 0.5]],
            model=SceneCraftModelConfig(
                use_lpips=True,
                use_l1=False,
                use_latent_loss=True,
                lpips_loss_mult=0.1,
                style_loss_mult=20.,
                latent_loss_mult=1,
                rgb_loss_mult=5.,
                interlevel_loss_mult=.5,
                distortion_loss_mult=0.002,
                depth_loss_mult=0.5,
                guide_depth_loss_mult=0.2,
                depth_consistency_loss_mult=5.,
                zvar_loss_mult=3.,
                use_appearance_embedding=True,
                use_direction_encoding=False,
                compute_background_color=False,
                eval_num_rays_per_chunk=1 << 15, # 32768
                train_num_rays_per_chunk=1 << 15, # force to be same as the backward process
                num_nerf_samples_per_ray=64,
                num_proposal_samples_per_ray=(512, 512),
                proposal_net_args_list=[
                    {"hidden_dim": 16, "log2_hashmap_size": 17, "num_levels": 5, "max_res": 512, "use_linear": False},
                    {"hidden_dim": 16, "log2_hashmap_size": 17, "num_levels": 7, "max_res": 2048, "use_linear": False},
                ],
                hidden_dim=256,
                hidden_dim_color=256,
                appearance_embed_dim=32,
                max_res=8192,
                proposal_weights_anneal_max_num_iters=20000,
                log2_hashmap_size=21,
                patch_size=224,
            ),
            checkpoint_path="gzzyyxy/layout_diffusion_hypersim_prompt_one_hot_multi_control_bs32_epoch24",
            checkpoint_subfolder="checkpoint-10900",
            # checkpoint_path="gzzyyxy/layout_diffusion_scannetpp_prompt_one_hot_multi_control_bs32_epoch18",
            # checkpoint_subfolder="checkpoint-15000",
            pretrained_model_name_or_path="stabilityai/stable-diffusion-2-1",
            guidance_use_full_precision=False,
        ),
        optimizers={
            "proposal_networks": {
                "optimizer": AdamOptimizerConfig(lr=1e-2, eps=1e-15, max_norm=10.),
                "scheduler": None,

            },
            "fields": {
                "optimizer": AdamOptimizerConfig(lr=1e-3, eps=1e-15, max_norm=10.),
                "scheduler": None,
            },
        },
        viewer=ViewerConfig(num_rays_per_chunk=1 << 15),
        vis="viewer",
    ),
    description="",
)
`   

At the last step 14999, the guided image (second column) generated by controlnet and the rendered image(third column) from nerf seem over-saturated

<img width="612" alt="Image" src="https://github.com/user-attachments/assets/a911880e-2ed8-472b-919b-374979eb9555" />

The loss curve of nerf fitting proocess is normal:

<img width="612" alt="Image" src="https://github.com/user-attachments/assets/dfe169d4-de30-4d34-a6fd-eb151920fd1b" />

Could you help debug this problem? @OrangeSodahub

**Comment by OrangeSodahub** [comment]
1. I'm a little confused, if you're training nerf, what is the 4th column of your shown image? Some recordings of my experiement is shown as below, there will only track 4 images: nerf, diffusion, rendered depth and bounding box depth.

![Image](https://github.com/user-attachments/assets/451f5371-bac2-4549-9e17-63980933fc67)

2. Nerf's generating results depends on various factors. To my knowledge,
    1) you could run more examples and adjust the parameters. For the smaller scene, set shorter training time (the `time_schedule` in config, (15K is too large for a scene in Hypersim, that is the direct reason of the over-saturated rgbs, I suggest 6K~10K for ~100 views);
    2) you could start from a simple room type, e.g. living room/bedroom, which will bring more consistent guidance from diffusion model; Some complex or not high-quality scenes (even from Hypersim itself) are not suitable (with non-regular objects represented by a bounding box, e.g. a round table, it causes ambiguity from different views). 
    3) try different prompts, but please make sure that the prompt is aligned with what you're try to generate. You used 'bedroom' but the layout is obviously not. In our experiments, matched prompt can also enhangce the results greatly, please refer to Figure F of our paper.

**Comment by OrangeSodahub** [comment]
If you have more issues, welcome to follow up.
